src/explore-education-statistics-common/src/modules/table-tool/contexts/TableHeadersContext.tsx

```diff
@@ -0,0 +1,109 @@
+import useToggle from '@common/hooks/useToggle';
+import noop from 'lodash/noop';
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+export interface TableHeadersContextState {
+  activeList?: string;
+  defaultNumberOfItems: number;
+  expandedLists: string[];
```

Comment:
Not keen on tracking expanded list states in context for the following reasons:

- Toggling any group will cause all groups to re-render - this is a little wasteful
- It creates unnecessary coupling to the context for determining if a group can be expanded/collapsed.
- This is more complicated than just passing state via props.

Couldn't we track this state for each group individually without using context?

Reply:
This was because when I just had it on each group then when they were reordered the state wasn't maintained correctly. For example, if I expanded Group 1, then moved it to after Group 2, Group 2 would then be expanded and Group 1 wouldn't be.
Can you suggest a better way to avoid this?

Reply:
This will be due to re-rendering groups in completely different parts of the DOM i.e. the components are being unmounted/re-mounted with completely fresh state.

You're right that there there probably isn't a way to maintain state locally! Happy to go with the current solution :+1: